src/Enum.php

```diff
@@ -296,9 +296,27 @@ private static function detectConstants($class)
     {
         if (!isset(self::$constants[$class])) {
             $reflection = new ReflectionClass($class);
-            $constants  = $reflection->getConstants();
+            $constants  = array();
+
+            do {
+                $scopeConstants = array();
+                if (PHP_VERSION_ID >= 70100) {
```

Comment:
using `version_compare()` is much much slower than using the integer constant as it first needs to call a function and the function has to parse three strings into two integers and a comparator sign. Instead I'm using the `PHP_VERSION_ID` integer constant to just do one comparison of two integers ;)